sunpy/visualization/animator/base.py

```diff
@@ -522,55 +500,112 @@ def _sanitize_axis_ranges(self, axis_ranges, data):
+                    extent = extent + [axis_ranges[i][0], axis_ranges[i][-1]]
+                    axis_ranges[i] = edges_to_centers_nd(axis_ranges[i])
+                elif axis_ranges[i].ndim == ndim and axis_ranges[i].shape[i] == data_shape[i]+1:
+                    extent = extent + [axis_ranges[i].min(), axis_ranges[i].max()]
+                    axis_ranges[i] = edges_to_centers_nd(axis_ranges[i], i)
+                else:
+                    raise ValueError(incompatible_axis_ranges_error_message(i))
+        # For each axis validate and translate the axis_ranges.
+        for i in self.slider_axes:
+            if axis_ranges[i] is None:
+                    axis_ranges[i] = np.arange(data_shape[i])
+            elif len(axis_ranges[i]) == 2:
+                axis_ranges[i] = np.linspace(axis_ranges[i][0], axis_ranges[i][-1],
+                                             data_shape[i]+1)
+                axis_ranges[i] = edges_to_centers_nd(axis_ranges[i])
+            elif len(axis_ranges[i]) == data_shape[i]+1:
+                # If array of individual pixel edges supplied, convert to pixel centers.
+                axis_ranges[i] = edges_to_centers_nd(np.asarray(axis_ranges[i]), i)
+            else:
+                raise ValueError(incompatible_axis_ranges_error_message(i))
 
-        return axis_ranges
+        return axis_ranges, extent
 
     @abc.abstractmethod
     def plot_start_image(self):  # pragma: no cover
         """
         Abstract method for plotting first slice of array.
 
-        Must exists here but be defined in subclass.
+        Must exist here but be defined in subclass.
 
         """
 
     @abc.abstractmethod
-    def update_plot(self):  # pragma: no cover
+    def update_plot(self, val, slider):  # pragma: no cover
         """
         Abstract method for updating plot.
 
-        Must exists here but be defined in subclass.
+        Must exist here but be defined in subclass.
 
         """
+        ind = int(val)
+        ax_ind = self.slider_axes[slider.slider_ind]
+        # Update slider label to reflect real world values in axis_ranges.
+        label = self.axis_ranges[ax_ind][ind]
+        slider.valtext.set_text("{0}".format(label))
+
+
+def edges_to_centers_nd(axis_range, edges_axis):
+    """
+    Converts ND array of pixel edges to pixel centers along one axis.
+    Parameters
+    ----------
+    axis_range: `numpy.ndarray`
+        Array of pixel edges.
+    edges_axis: `int`
+        Index of axis along which centers are to be calculated.
+    """
+    upper_edge_indices = [slice(None)] * axis_range.ndim
+    upper_edge_indices[edges_axis] = slice(1, axis_range.shape[edges_axis])
+    upper_edges = axis_range[tuple(upper_edge_indices)]
+    
+    lower_edge_indices = [slice(None)] * axis_range.ndim
+    lower_edge_indices[edges_axis] = slice(0, -1)
+    lower_edges = axis_range[tuple(lower_edge_indices)]
+    
+    return (upper_edges - lower_edges) / 2 + lower_edges
```

Comment:
We do not need to worry about any rounding here?

Reply:
No.  This PR supports non-integer edge labels.